### ISSUE TITLE
Focus Point ignored with Eager Loading

### ISSUE THREAD

**Issue by terryupton** [aelvan/Imager-Craft#168]
Hi @aelvan - not sure exactly where this issue lies, with FocalPoint or with Imager. However, it appears there is an issue when combining Focus Point and Imager when eager loading.
Take the example code;

```
{% set news = craft.entries({
  section: 'news',
  with: ['articleImage'],
  limit: limit,
}) %}

    {% spaceless %}
      {% set image = entry.articleImage[0] ?? fallbackBannerImage.bannerImage[0] ?? null  %}
    {# IMAGE TRANSFORMS #}
      {% set alt = (image.alt ?: entry.title) %}
      {# Set the ratio for the image crops #}
      {% set imageRatioWidth = 16 %}
      {% set imageRatioHeight = 8 %}
      {% set mode = 'crop' %}

      X = {{ image.focusPctX }} and Y = {{ image.focusPctY }}

      {% set srcImage = craft.imager.transformImage(image, { mode: mode, width:400, ratio: imageRatioWidth/imageRatioHeight, position: image.focusPctX ~ '% ' ~ image.focusPctY ~ '%', format: 'jpg' }) %}

      {# Set the responsive sizes based on the original image size based on the array passed into the Macro. #}
      {% set widths = [1600, 1200, 900, 600, 400, 200] %}
      {% set imageSizes = [] %}
      {% for width in widths %}
        {% if (image.width > width) %}
            {% set imageSizes = imageSizes|merge([{ 'width': width }]) %}
        {% endif %}
      {% endfor %}

      {# Set Srcset Transform #}
      {% set transformedImages = craft.imager.transformImage(image, imageSizes, { mode: mode, ratio: imageRatioWidth/imageRatioHeight, position: image.focusPctX ~ '% ' ~ image.focusPctY ~ '%', format: 'jpg' }) %}
    {% endspaceless %}
```

This results in cropping 50% and 50% - I can see that the FocusPoint values still exist when I access them via {{ image.focusPctX }} and Y = {{ image.focusPctY }} (as above). 

If I remove the with (Eager Loading) then the values are respected and the image is cropped correctly.

Thanks for your time.

**Comment by terryupton** [comment]
I should also add this was raised here too. https://github.com/smcyr/Craft-FocusPoint/issues/22

**Comment by aelvan** [comment]
What happens if you try to do `{{ image.focusPctX ~ '% ' ~ image.focusPctY ~ '%' }}` just before the last transformImage call?

**Comment by terryupton** [comment]
Changing this placement doesn't seem to have any bearing. Oddly though for some images it isn't always set to 50% 50%.

I have one example that with Eager Loading = off the points are correct and read as:
23.5% 64.5%

But with it on it reads as:
50.5% 52%

Even more odd is that for one image it gets the right values regardless of one or off. I thought for a second it might have been caching the first focusPoint set on the image, but I ruled this out by changing the working example and it updated both with the Eager Loading on and off. But only this image worked. It was number 3 in the loop (53.5% 31.5%).

I have attached a few screenshots if it is of use? I can always hook up a staging version if you wish to insect the code.

Thanks,

![Eager Loading = On](https://user-images.githubusercontent.com/5698596/39329409-21e1736a-4996-11e8-85e3-414cc7350479.png)

![Eager Loading = Off](https://user-images.githubusercontent.com/5698596/39329490-60c5e520-4996-11e8-918a-30a2010627be.png)
